packages/ripple-binary-codec/src/binary.ts

```diff
@@ -105,13 +106,19 @@ interface ClaimObject extends JsonObject {
  * @returns the serialized object with appropriate prefix
  */
 function signingClaimData(claim: ClaimObject): Buffer {
-  const num = bigInt(String(claim.amount))
   const prefix = HashPrefix.paymentChannelClaim
   const channel = coreTypes.Hash256.from(claim.channel).toBytes()
-  const amount = coreTypes.UInt64.from(num).toBytes()
-
+  if (typeof claim.amount === 'object') {
```

Comment:
This change involves a lot of repeated code, such as `bytesList.put(prefix)`. Please refactor to clean it up, so the only thing in the if/else is the processing of the amount.